src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/InputPlaybackActionData.cs

```diff
@@ -51,10 +52,19 @@ public bool ProcessAction(int segmentNumber, Dictionary<long, ObjectStatus> curr
 
                 var allInputs = inputData.AllInputsSortedByTime();
 
+                HashSet<Key> keysSeenThisUpdate = new();
+                bool stopKeyboard = false;
+
                 foreach (object input in allInputs)
                 {
-                    if (input is KeyboardInputActionData replayKeyboardInputEntry)
+                    if (!stopKeyboard && input is KeyboardInputActionData replayKeyboardInputEntry)
```

Comment:
I am potentially missing something or a little bit confused - wouldn't this stop any other key events from being sent as well? For instance, if my actions are press A, A, then B, wouldn't B fail to ever be executed?

Reply:
Ah I see... that makes more sense now. Thanks!